R/string_operation.R

```diff
@@ -216,30 +232,30 @@ stem_word <- function(...){
 #' @param df Data frame which has tokens.
 #' @param token Column name of token data.
 #' @param n How many tokens should be together as new tokens. This should be numeric vector.
-#' @param skip How many tokens can be skipped when connecting them.
 #' @export
-do_ngram <- function(df, token, n=1:2, skip=0){
+do_ngram <- function(df, token, sentence, document, maxn=2, sep="_"){
   loadNamespace("dplyr")
   loadNamespace("tidyr")
-  loadNamespace("quanteda")
+  loadNamespace("stringr")
+  loadNamespace("lazyeval")
   token_col <- col_name(substitute(token))
+  sentence_col <- col_name(substitute(sentence))
+  document_col <- col_name(substitute(document))
 
-  indices <- attr(df, "indices")
-  quanteda::ngrams
-  if(is.null(indices)){
-    # not grouped case
-    skipgrams <- quanteda::skipgrams(as.character(df[[token_col]]), n=n, skip=skip)
-    ret <- data.frame(skipgram=skipgrams, stringsAsFactors = F)
-    colnames(ret) <- token_col
-    ret
-  } else {
-    # grouped case
-    labels <- attr(df, "labels")
-    labels[[token_col]] <- lapply(indices, function(index){
-      quanteda::skipgrams(as.character(df[[token_col]][index+1]), n=n, skip=skip)
-    })
-    tidyr::unnest_(labels, token_col)
+  grouped <- (df %>%
+            dplyr::group_by_(document_col, sentence_col))
+  prev_cname <- token_col
+  for(n in seq(maxn)[-1]){
```

Comment:
Can you add some more comments on how ngram is done here? I found it hard to figure out without comment.